lib/message.rb

```diff
@@ -60,12 +60,19 @@ def as_json(options = {})
 
       if not @labels.nil? and @labels != []
         hash["label_ids"] = @labels.map do |label|
+          if !label.respond_to?(:id)
+            raise TypeError, "label #{label} does not respond to #id"
+          end
           label.id
         end
       end
 
-      if not @folder.nil?
-        hash["folder_id"] = @folder.id
+      if !folder.nil? && !folder.respond_to?(:id)
+        raise TypeError, "folder #{folder} does not respond to #id"
+      end
+
+      if !folder.nil? && folder.respond_to?(:id)
+        hash["folder_id"] = folder.id
       end
```

Comment:
We may want to pomote this to `ModelState#as_json` later on.

Reply:
Hmm, do you mean `RestfulModel`?

Reply:
Ah, no. model_state is a class that's job is to track whether or not the data has mutated. It's part fo the branch that we will be merging (hopefully soon) to ensure we only send the changed data when posting to the API to reduce the prevalance of "overwrite" bugs.

Reply:
Here's the sprouted RestfulModel class: https://github.com/nylas/nylas-ruby/pull/113/files#diff-a351f8985b331720cad0220e410159beR9